discord-scripts/discord-webhook.ts

```diff
@@ -51,25 +51,82 @@ export default async function webhookDiscord(
     }
   }
 
+  async function getUserIdByName(
+    guildId: string,
+    username: string,
+  ): Promise<string | null> {
+    const guild = discordClient.guilds.cache.get(guildId)
+    if (!guild) throw new Error("Guild not found")
+
+    await guild.members.fetch()
+    // WIP, output list of all members for matching
+    guild.members.cache.forEach((member) => {
+      robot.logger.info(`Username: ${member.user.username}`)
+    })
+
+    const matchedMember = guild.members.cache.find((member) =>
+      member.user.username.includes(username),
+    )
+
+    return matchedMember ? matchedMember.user.id : null
+  }
+
+  async function updateServerNickname(
```

Comment:
This isn't just about updating the server nickname since it does all the OOO stuff, right? Let's be very specific about what this function does in the name.